src/geode/geometry/intersection.cpp

```diff
@@ -170,11 +168,9 @@ namespace geode
             line_plane_intersection( InfiniteLine3D{ segment }, plane );
         if( line_plane_result )
         {
-            double distance_to_segment;
-            std::tie( distance_to_segment, std::ignore ) =
-                point_segment_distance(
-                    line_plane_result.result.value(), segment );
-            if( distance_to_segment > global_epsilon )
+            if( new_point_segment_distance(
```

Comment:
Tu voulais pas aussi remplacer l'usage des coord barycentric dans les correctness?

Reply:
Yes I will do it in a next PR